state/state.go

```diff
@@ -27,3 +27,41 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+type Key struct {
+	Name string
+	Mode [ModeLen]byte
+}
+
+const (
+	ModeLen = 8
```

Comment:
We are probably fine with getting away with a hard-coded value/a single byte but should consider a dynamically sized []byte as an alternative